java-checks/src/main/java/org/sonar/java/checks/TooManyParametersCheck.java

```diff
@@ -66,33 +63,67 @@ public class TooManyParametersCheck extends BaseTreeVisitor implements JavaFileS
     "javax.ws.rs.PUT",
     "javax.ws.rs.PATCH",
     "org.springframework.beans.factory.annotation.Autowired",
-    "javax.inject.Inject"
+    "javax.inject.Inject",
+    "org.springframework.context.annotation.Bean",
+    "io.micronaut.http.annotation.Get",
+    "io.micronaut.http.annotation.Post",
+    "io.micronaut.http.annotation.Put",
+    "io.micronaut.http.annotation.Delete",
+    "io.micronaut.http.annotation.Options",
+    "io.micronaut.http.annotation.Patch",
+    "io.micronaut.http.annotation.Head",
+    "io.micronaut.http.annotation.Trace"
   );
+  
+  //if a class is annotated as one of these types, its constructor should be ignored
+  private static final List<String> CLASS_CONSTRUCTOR_WHITELIST = Arrays.asList(
```

Comment:
When we modify the existing code, it's a good opportunity to replace `loaded terms` like `blacklist/whitelist`, `master/slave`. So here, I propose to replace `WHITE_LIST` with `METHOD_ANNOTATION_EXCEPTIONS` and `CLASS_CONSTRUCTOR_WHITELIST` with `CLASS_ANNOTATION_IMPLYING_CONSTRUCTOR_EXCEPTIONS` ... or something smaller ...

Reply:
I updated the names of those static final lists accordingly